test/unit_tests/service/grid/grid_service_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:f1_application/lib/model/driver.dart';
+import 'package:f1_application/lib/service/grid/grid_service.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:collection/collection.dart';
+
+void main() {
+  final GridService service = GridService();
+
+  final List<Driver> drivers = [
+    Driver(id: "1", wikiURL: "https://wikipedia.com", firstName: "One", lastName: "Driver", dateOfBirth: "2001-01-01", nationality: "Italian", code: "DON", permanentNumber: "01"),
+    Driver(id: "2", wikiURL: "https://wikipedia.com", firstName: "Two", lastName: "Driver", dateOfBirth: "2002-01-01", nationality: "Brazilian", code: "DTW", permanentNumber: "02"),
+    Driver(id: "3", wikiURL: "https://wikipedia.com", firstName: "Three", lastName: "Driver", dateOfBirth: "2003-01-01", nationality: "Columbian", code: "DTH", permanentNumber: "03"),
+    Driver(id: "4", wikiURL: "https://wikipedia.com", firstName: "Four", lastName: "Driver", dateOfBirth: "2004-01-01", nationality: "Hungarian", code: "DFO", permanentNumber: "04"),
+    Driver(id: "5", wikiURL: "https://wikipedia.com", firstName: "Five", lastName: "Driver", dateOfBirth: "2005-01-01", nationality: "Brazilian", code: "DFI", permanentNumber: "05"),
+  ];
+
+  group("Grid Service test(s)", () {
+    group("relevantDriversByExpression function test(s)", () {
+      group("With drivers List", () {
+        test("Get all driver with no expression ''", () {
+          final result = service.relevantDriversByExpression(drivers, "");
+          expect(const ListEquality().equals(result, drivers), true);
+        });
+
+        test("Get 'One Driver' driver with expression 'One'", () {
+          final result = service.relevantDriversByExpression(drivers, "One");
+          expect((
+              result.contains(drivers[0]) && result.length == 1

```

Comment:
Ha van contains, `result.length == 1` -re már nem vizsgálnék, mert ha bővül a teszt adat (pl. bekerül még valaki akinek a nevében "One" szerepel), akkor ez a teszt el fog romlani

Ez több helyen is van típushibaként, visszatörölgetheted őket